### DISCUSSION TITLE
How can I enable only the copy part of OSC52?

### DISCUSSION THREAD

**Discussion by MasterAwesome** [neovim/neovim#27500] [Q&A] [answered]
With NVIM v0.10 I am able to osc-52 to copy text across ssh although the paste gets stuck in _"Waiting for OSC 52 response from terminal"_ and gets stuck there until it times out. 

I want to only enable the copy part of OSC52 or alternatively use vim registers for copy and paste unless I specify the `+` register. How can I achieve this?

I'm using NVChad and enabling OSC52 by setting `g.clipboard` as per `:help clipboard`

**Comment by stseifer** [comment]
I am also experiencing the same issue

``` System Information: {
    machine = "x86_64",
    release = "6.6.9-arch1-1",
    sysname = "Linux",
    version = "#1 SMP PREEMPT_DYNAMIC Tue, 02 Jan 2024 02:28:28 +0000"
  }
  nvim version: 0.10.0-dev+g185752614
  Terminal: wezterm
  using nvim over ssh
  ```
  
 There must be something that makes this behavior happen in our config because running `nvim --clean`  works as expected using `"+y`
 
 Edit: doing more research this happens when pressing `"` this prompts which-key plugin. disabling which-key osc52 works as expected

**Comment by brianhuster** [comment, marked as answer]
Example using OSC52 for copying, and powershell for pasting

```lua
vim.g.clipboard = {
  name = 'myclipboard',
  copy = {
    ['+'] = require('vim.ui.clipboard.osc52').copy('+'),
    ['*'] = require('vim.ui.clipboard.osc52').copy('*'),
  },
  paste = {
    ['+'] = 'powershell.exe -NoLogo -NoProfile -c [Console]::Out.Write($(Get-Clipboard -Raw).tostring().replace("`r", ""))'
    ['*'] = 'powershell.exe -NoLogo -NoProfile -c [Console]::Out.Write($(Get-Clipboard -Raw).tostring().replace("`r", ""))'
  },
}
```
